File: solutions/python3/0128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        set_ = set(nums)

        for num in nums:
            if num - 1 not in set_:
                length = 0
                while num in set_:
                    num += 1
                    length += 1
                ans = max(ans, length)

        return ans

File: solutions/java/0128.java
class Solution {
  public int longestConsecutive(int[] nums) {
    int ans = 0;

    Set<Integer> set = new HashSet<>();
    for (int num : nums)
      set.add(num);

    for (int num : nums)
      if (!set.contains(num - 1)) {
        int length = 0;
        while (set.contains(num++))
          ++length;
        ans = Math.max(ans, length);
      }

    return ans;
  }
}